### PR TITLE
feat(ticket): allow project members to view  and owners to update all tickets in the project

### PR DESCRIPTION
**Pull request by MarkPotato777**
#### What type of PR is this?

type-feature
#### What this PR does / why we need it:
1. allow project members to view  and owners to update all tickets in the project
2. support filter by projects in the tickets list
3. refactor flow instance permission check logic

### CODE REVIEW

**Review by PeachThinking — DISMISSED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/metadb/flow/FlowInstanceSpecs.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public static Specification<FlowInstanceEntity> parentInstanceIdIn(Collection<Lo
         return SpecificationUtil.columnIn(FLOW_INSTANCE_PARENT_INSTANCE_ID, parentInstanceId);
     }
 
-    public static Specification<FlowInstanceEntity> projectIdEquals(Long projectId) {
```
[the remark this answers is not in the document]
Comment (MarkPotato777) [reply server/odc-service/src/main/java/com/oceanbase/odc/metadb/flow/FlowInstanceSpecs.java:101]: The method has no reference now, I prefer to delete it directly.

**Review by guowl3 — DISMISSED** [review]

**Review by guowl3 — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by PeachThinking — APPROVED** [review]

**Review by PeachThinking — APPROVED** [review]

**Review by yizhouxw — APPROVED** [review]
lgtm

**Pull request merged**